File: apps/api/app/schemas/generate.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class ResumeSuggestions(BaseModel):
    emphasize: list[str] = []
    new_bullets: list[str] = []
    professional_summary: str = ""


class CoverLetterResponse(BaseModel):
    content: str
    job_id: uuid.UUID


class EmailResponse(BaseModel):
    content: str
    job_id: uuid.UUID


class RoadmapItem(BaseModel):
    skill: str
    importance: str = ""
    resources: list[str] = []
    project_idea: str = ""
    estimated_time: str = ""


class RoadmapResponse(BaseModel):
    items: list[RoadmapItem]
    job_id: uuid.UUID


class GeneratedOutputResponse(BaseModel):
    id: uuid.UUID
    job_id: uuid.UUID
    output_type: str
    content: str
    created_at: datetime

    model_config = {"from_attributes": True}

File: apps/web/src/lib/api/generate.ts
import api from './client';

export interface ResumeSuggestions {
  emphasize: string[];
  new_bullets: string[];
  professional_summary: string;
}

export interface CoverLetterResponse {
  content: string;
  job_id: string;
}

export interface EmailResponse {
  content: string;
  job_id: string;
}

export interface RoadmapItem {
  skill: string;
  importance: string;
  resources: string[];
  project_idea: string;
  estimated_time: string;
}

export interface PrepQuestion {
  category: string;
  question: string;
  suggested_answer: string | null;
  difficulty: string | null;
}

export interface PrepResponse {
  job_id: string;
  technical: PrepQuestion[];
  behavioral: PrepQuestion[];
  system_design: PrepQuestion[];
}

export async function getResumeSuggestions(jobId: string): Promise<ResumeSuggestions> {
  const { data } = await api.post(`/generate/resume-suggestions/${jobId}`);
  return data;
}

export async function getCoverLetter(jobId: string): Promise<CoverLetterResponse> {
  const { data } = await api.post(`/generate/cover-letter/${jobId}`);
  return data;
}

export async function getRecruiterEmail(jobId: string): Promise<EmailResponse> {
  const { data } = await api.post(`/generate/recruiter-email/${jobId}`);
  return data;
}

export async function getRoadmap(jobId: string): Promise<{ items: RoadmapItem[]; job_id: string }> {
  const { data } = await api.post(`/generate/roadmap/${jobId}`);
  return data;
}

export async function getPrepQuestions(jobId: string): Promise<PrepResponse> {
  const { data } = await api.post(`/prep/questions/${jobId}`);
  return data;
}
